server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/lib/AbstractStateMailetProcessorTest.java

```diff
@@ -198,6 +200,7 @@ public void afterMailet(Mailet m, String mailName, String state, long processTim
 
         // the source mail should have state error as the exception was thrown
         assertEquals(Mail.ERROR, mail.getState());
+        assertEquals(MessagingException.class, mail.getAttribute(Mail.MAILET_ERROR_ATTRIBUTE_NAME));
```

Comment:
this assertions does not belong to this test because as the test name says : "matcher Processing Should Not Result In An Exception When Matcher Throws" but not "matcher processing should capture exception as an email attribute".

Would you mind adding a new test for that ?